src/SprykerEco/Service/ComputopApi/ComputopApiService.php

```diff
@@ -64,12 +64,12 @@ public function extractResponseHeader(array $plaintextResponseHeader, $method):
         $header = $this->getFactory()->createComputopApiConverter()->extractResponseHeader($plaintextResponseHeader, $method);
 
         $expectedMac = $this->getHashValue(
-            $this->getFactory()->createComputopApiMapper()->getMacResponseEncryptedValue($header)
+            $this->getFactory()->createComputopApiMapper()->getMacResponseEncryptedValue($header),
         );
-        $this
-            ->getFactory()
+
+        $this->getFactory()
             ->createComputopApiConverter()
-            ->checkMacResponse($header->getMac(), $expectedMac, $header->getMethod());
+            ->checkMacResponse($header->getMacOrFail(), $expectedMac, $header->getMethodOrFail());
```

Comment:
Please, mention in the method's specification that these fields are required.

Reply:
Fixed